frontend/src/bundles/common/components/stepper/stepper.tsx

```diff
@@ -0,0 +1,59 @@
+import { IconName } from '../../icons/icons.js';
+import { Box, Flex, Header, Icon, IconButton } from '../components.js';
+import { Step } from './components/components.js';
+import { getStepStatus } from './helpers/helpers.js';
+
+type Properties = {
+    steps: string[];
+    currentStep: string;
+};
+
+const stepsElements: React.FC<Properties> = ({ steps, currentStep }) => {
+    return steps.map((stepName, index) => {
+        return (
+            <Step
+                key={index}
+                stepDescription={stepName}
+                step={getStepStatus({ stepName, index, currentStep, steps })}
+            />
+        );
+    });
+};
+
+const Stepper: React.FC<Properties> = ({ steps, currentStep }) => {
```

Comment:
Btw, did you consider using chakra Stepper instead of writing it from scratch?

Reply:
That Stepper does not support the horizontal layout we need. As a result, I had to implement this feature myself while ensuring that the spacing between steps and their descriptions is maintained.